### DISCUSSION TITLE
20230921 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#60] [Meetings]
Location:
<https://meeting.hodl10.io/ConsistentClassroomsMergeSideways>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x 
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Index

Small issues:

- [#[273](https://github.com/torrust/torrust-index/issues/273)] https://github.com/torrust/torrust-index/issues/273
- [#[288](https://github.com/torrust/torrust-index/issues/288)] https://github.com/torrust/torrust-index/issues/288
- [#[289](https://github.com/torrust/torrust-index/issues/289)] https://github.com/torrust/torrust-index/issues/289
- [#[302](https://github.com/torrust/torrust-index/issues/302)] https://github.com/torrust/torrust-index/issues/302

Refactorings:

- [#[276](https://github.com/torrust/torrust-index/issues/276)] https://github.com/torrust/torrust-index/issues/276

Improved (I hope) version of the [services::Torrent::Index::add_torrent](https://github.com/torrust/torrust-index/blob/develop/src/services/torrent.rs#L127-L178) function.

Bugs:

- [#[266](https://github.com/torrust/torrust-index/issues/266)] https://github.com/torrust/torrust-index/issues/266

I found the [problem](https://github.com/torrust/torrust-index/issues/266#issuecomment-1721176747). 

Decode fails in serde_bencode when there is a list with more than one node:

```json
{
   "info": {
      "length": 8,
      "name": "minimal.txt",
      "piece length": 16384,
      "pieces": "<hex>30 9A 84 51 4A F1 09 D0 8C 34 42 11 26 67 7F 84 B3 23 4D 78</hex>"
   },
   "nodes": [
      [
         "188.163.121.224",
         56711
      ],
      [
         "162.250.131.26",
         13386
      ]
   ]
}
```

I do not know what the generic problem is if it's a list with more than one element or a list inside a dict, ... I've opened a [PR](https://github.com/toby/serde-bencode/pull/32), but they have not replied. Should we fork, fix, and publish our crate?

Waiting for feedback:

- https://github.com/torrust/torrust-index/issues/290

**Comment by da2ce7** [comment]
## Torrust Tracker Release Process
I completed the release process document for the torrust tracker.
<https://github.com/torrust/torrust-tracker/blob/develop/docs/release_process.md>

## Tested the process with a release:
- https://github.com/torrust/torrust-tracker/issues/456

<https://github.com/torrust/torrust-tracker/releases/tag/v3.0.0-alpha.11>
<https://crates.io/crates/torrust-tracker/3.0.0-alpha.11>
<https://hub.docker.com/layers/torrust/tracker/v3.0.0-alpha.11/images/sha256-c5980f0d4ae0b63d3a9dad05d494af9badf0833892f5bb066b3f673077a5eb9b?context=explore>

## Torrust Tracker GitHub Rullsets
I completed the rulesets for the torrust tracker:
<https://github.com/torrust/torrust-tracker/rules>

## Renamed Torrust Repos:
`torrust-index-backend` => `torrust-index`
`torrust-index-frontend` => `torrust-index-gui`

## Renamed the Updated Nautilus Github:
`Nautilus-Cyberneering` => `nautilus-cyberneering`

## Worked on the Nautilus Github Profile:
- https://github.com/nautilus-cyberneering/.github/pull/58

**Comment by mario-nt** [comment]
**Front**

- Closed https://github.com/torrust/torrust-index-gui/issues/257

**Index**

- Worked on https://github.com/torrust/torrust-index-backend/issues/285
- Worked on https://github.com/torrust/torrust-index-backend/issues/284
- Worked on https://github.com/torrust/torrust-index-backend/issues/39
- Commented on and researched about https://github.com/torrust/torrust-index-backend/issues/301
- Commented on and researched about https://github.com/torrust/torrust-index/discussions/297


**Torrust index api lib and types lib**

- Closed https://github.com/torrust/torrust-index-api-lib/issues/33#event-10384349336 with @josecelano 

**Other**

- Commented on and researched about https://github.com/orgs/torrust/discussions/13

**Training**

- Research on Axum (handlers, extractors, middleware, tower, axum)
- Research on Rust (ARC, Option, match, #must_use, where...)

**Comment by grmbyrn** [comment]
## SvelteKit Blog/Secure Git Guide
* Made a blog using SK, TypeScript, and Markdown to get a better handle on how to use them together. Repo available here: https://github.com/grmbyrn/sk-blog
* Implemented lessons learned from this in the Secure Git Guide, mainly through styling Markdown used in Svelte components.

## Research/Study
* Course on TypeScript in order to implement it in current project, but also in future projects with and without SvelteKit.
* Daisy UI as recommended in meeting with Jose for use in current project. Tailwind CSS, which Daisy UI is a plugin for, seems to cancel Markdown syntax so used Open Props for now.
* Revised more advanced Git commands.
* Confirmed that I can buy credit with OpenAI to access their API key in order to work on AI projects.
